au/code/au/math_test.cc

```diff
@@ -159,6 +159,60 @@ TEST(clamp, SupportsZeroForMultipleArguments) {
     EXPECT_THAT(clamp(feet(6), ZERO, ZERO), SameTypeAndValue(feet(0)));
 }
 
+TEST(hypot, QuantityConsistentWithStdHypotWhenTypesAreIdentical) {
+    auto expect_consistent_with_std_hypot = [](auto u, auto v) {
+        const auto expected = ohms(std::hypot(u, v));
+        const auto actual = hypot(ohms(u), ohms(v));
+        EXPECT_THAT(actual, SameTypeAndValue(expected));
+    };
+
+    // Rep: `int`.
+    expect_consistent_with_std_hypot(-1, 0);
+    expect_consistent_with_std_hypot(0, 0);
+    expect_consistent_with_std_hypot(1, 0);
+    expect_consistent_with_std_hypot(2, 0);
+    expect_consistent_with_std_hypot(4, 2);
+
+    // Rep: `double`.
+    expect_consistent_with_std_hypot(-1.0, 0.0);
+    expect_consistent_with_std_hypot(0.0, 0.0);
+    expect_consistent_with_std_hypot(1.0, 0.0);
+    expect_consistent_with_std_hypot(2.0, 0.0);
+    expect_consistent_with_std_hypot(4.0, 2.0);
+}
+
+TEST(hypot, QuantityProducesResultsInCommonUnitOfInputs) {
+    EXPECT_THAT(hypot(centi(meters)(30), milli(meters)(400)),
+                SameTypeAndValue(milli(meters)(500.0)));
+}
```

Comment:
There's a really cute test case you could consider adding here, that takes advantage of the pythagorean triple (5, 12, 13), and the fact that 1 ft == 12 in.  We could justify it by adding coverage for a non-`double` rep:

```suggestion

    EXPECT_THAT(hypot(inches(5.f), feet(1.f)), SameTypeAndValue(inches(13.f)));
}
```